packages/paste-core/components/code-block/src/CodeBlockTabList.tsx

```diff
@@ -17,12 +77,67 @@ export interface CodeBlockTabListProps extends Omit<TabListProps, "aria-label">
 
 export const CodeBlockTabList = React.forwardRef<HTMLDivElement, CodeBlockTabListProps>(
   ({ children, element = "CODE_BLOCK_TAB_LIST", ...props }, ref) => {
+    const tab = React.useContext(TabsContext);
+    const theme = useTheme();
+    //  ref to the scrollable element
+    const scrollableRef = React.useRef<HTMLDivElement>(null);
+    const [scrollShadow, setScrollShadow] = React.useState<"none" | "left" | "right" | "both">("none");
+
+    // Function to handle scroll event
+    const handleScroll = (): void => {
+      if (scrollableRef.current) {
+        // No scrollbar, so no shadow
+        if (scrollableRef.current.clientWidth === scrollableRef.current.scrollWidth) {
```

Comment:
suggestion: can you use refElement.getBoundingClientRect() here instead? hopefully it will provide everything you need and I _think_ it's supposed to be more accurate to what's actually rendered in the DOM

https://stackoverflow.com/questions/32438642/clientwidth-and-clientheight-report-zero-while-getboundingclientrect-is-correct

Reply:
Great suggestion. I made the change and is working well